From fd72b361f3092390f0b3f7758f6a7b1d9b5fddcf Mon Sep 17 00:00:00 2001
From: Michael Lee <michael-cy.lee@mediatek.com>
Date: Tue, 11 Jul 2023 14:17:43 +0800
Subject: [PATCH 040/149] mtk: hostapd: Set WMM and TX queue parameters for
 wpa_supplicant

Since most of the time, wpa_supplicant will be used to setup an STA
interface, it's default WMM and TX queue parameters should be set for
STA.

Signed-off-by: Michael Lee <michael-cy.lee@mediatek.com>
---
 wpa_supplicant/config.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/wpa_supplicant/config.c b/wpa_supplicant/config.c
index 8d5df3590..895dab4b0 100644
--- a/wpa_supplicant/config.c
+++ b/wpa_supplicant/config.c
@@ -4724,19 +4724,19 @@ struct wpa_config * wpa_config_alloc_empty(const char *ctrl_interface,
 	const struct hostapd_wmm_ac_params ac_bk =
 		{ aCWmin, aCWmax, 7, 0, 0 }; /* background traffic */
 	const struct hostapd_wmm_ac_params ac_be =
-		{ aCWmin, aCWmax, 3, 0, 0 }; /* best effort traffic */
+		{ aCWmin, aCWmin + 2, 3, 0, 0 }; /* best effort traffic */
 	const struct hostapd_wmm_ac_params ac_vi = /* video traffic */
-		{ aCWmin - 1, aCWmin, 2, 3008 / 32, 0 };
+		{ aCWmin - 1, aCWmin, 1, 3008 / 32, 0 };
 	const struct hostapd_wmm_ac_params ac_vo = /* voice traffic */
-		{ aCWmin - 2, aCWmin - 1, 2, 1504 / 32, 0 };
+		{ aCWmin - 2, aCWmin - 1, 1, 1504 / 32, 0 };
 	const struct hostapd_tx_queue_params txq_bk =
 		{ 7, ecw2cw(aCWmin), ecw2cw(aCWmax), 0 };
 	const struct hostapd_tx_queue_params txq_be =
-		{ 3, ecw2cw(aCWmin), 4 * (ecw2cw(aCWmin) + 1) - 1, 0 };
+		{ 3, ecw2cw(aCWmin), ecw2cw(aCWmax), 0 };
 	const struct hostapd_tx_queue_params txq_vi =
-		{ 1, (ecw2cw(aCWmin) + 1) / 2 - 1, ecw2cw(aCWmin), 30 };
+		{ 2, (ecw2cw(aCWmin) + 1) / 2 - 1, ecw2cw(aCWmin), 30 };
 	const struct hostapd_tx_queue_params txq_vo =
-		{ 1, (ecw2cw(aCWmin) + 1) / 4 - 1,
+		{ 2, (ecw2cw(aCWmin) + 1) / 4 - 1,
 		  (ecw2cw(aCWmin) + 1) / 2 - 1, 15 };
 
 #undef ecw2cw
-- 
2.45.2

